File: client/src/app/_models/params/callRecordParams.ts
export interface ICallRecordParams
{
    id: number;
    personType: string;
    incomingOutgoing: string;
    dateOfContact: Date;
    personId: string;
    mobileNo: string;
    emailId: string;
    categoryRef:string;
    status: string;
    statusClass: string;
    subject: string;
    username: string;
    gistOfDiscussions: string;
    advisoryBy: string;
    nextAction:string;
    nextActionOn: Date;

    search: string;
    sort: string;
    pageNumber: number;
    pageSize: number;
}


export class CallRecordParams implements ICallRecordParams
{
    id = 0;
    personType = '';
    incomingOutgoing='';
    dateOfContact=new Date();
    personId="";
    mobileNo = '';
    emailId = '';
    categoryRef = '';
    status = '';
    statusClass='Active';
    username =  '';
    gistOfDiscussions='';
    advisoryBy='';
    nextAction='';
    nextActionOn=new Date();
    subject='';

    search = '';
    sort = '';
    pageNumber=1;
    pageSize=10;
}

File: api/Params/Admin/CallRecordParams.cs
namespace api.Params.Admin
{
    public class CallRecordParams: PaginationParams
    {
        public int Id { get; set; }
        public string IncomingOutgoing { get; set; }
        public string Email { get; set; }
        public string MobileNo { get; set; }
        public string PersonType { get; set; }
        public string PhoneNo { get; set; }
        public string PersonId { get; set; }
        public string Subject { get; set; }
        public string Status { get; set; }
        public string StatusClass { get; set; }
        public string Username { get; set; }
        public string CategoryRef { get; set; }
        public DateTime DateRegistered { get; set; }
        public string AdvisoryBy { get; set; }
        public string GistOfDiscussions { get; set; }

    }
}